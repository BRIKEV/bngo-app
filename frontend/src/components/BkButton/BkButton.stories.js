import { storiesOf } from '@storybook/vue';

import BkButton from './BkButton.vue';

storiesOf('Components/BkButton', module)
  .add('default', () => ({
    components: { BkButton },
    template: `
      <BkButton :isLoading="false">Login</BkButton>
        `,
  }));
