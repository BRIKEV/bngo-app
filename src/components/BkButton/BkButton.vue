<template>
  <button
    :class="{ outline }"
    class="btn"
    :disabled="disabled"

    @click="btnClicked">
    <beat-loader :loading="isLoading" :color="'white'" :size="10" />
    <p class ="content" v-if="!isLoading">
      <slot></slot>
    </p>
  </button>
</template>

<script>
export default {
  name: 'BkButton',

  props: {
    isLoading: {
      type: Boolean,
      default: false,
    },
    disabled: Boolean,
    outline: {
      type: Boolean,
      default: false,
    },
  },

  methods: {
    btnClicked() {
      this.$emit('btn-clicked');
    },
  },

};
</script>

<style lang="scss" scoped>
@import "@/theme/index.scss";

.btn {
  background: $brand;
  border-radius: calculateRem(8px);
  color: $white;
  min-height: calculateRem(45px);
  text-align: center;
  text-decoration: none;
  font-weight: bold;
  font-size: $fs-base;
  transition: all .15s linear;
  padding: 5px 15px;
  cursor: pointer;
  &:hover {
    background: lighten($brand, 20%);
    transition: background .3s ease-in;
  }

  &:active {
    background-color: $brand;
  }

  &:disabled {
    cursor: not-allowed;
    background: lightgrey;
  }

  &.outline {
    background: $white;
    color: $brand;
    border-color: $brand;
    &:hover {
      background: lighten($brand, 20%);
      color: $white;
      border-color: inherit;
      transition: background .3s ease-in;
    }
  }

  .content {
    margin: 0;
  }
}
</style>
