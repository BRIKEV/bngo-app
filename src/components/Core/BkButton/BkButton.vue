<template>
  <button
    :class="{ outline }"
    class="btn"
    :disabled="disabled"

    @click="btnClicked">
    <beat-loader :loading="isLoading" :color="'white'" :size="10" />
    <p class ="content" v-if="!isLoading">
      <slot></slot>
    </p>
  </button>
</template>

<script>
import VueTypes from 'vue-types';

export default {
  name: 'BkButton',

  props: {
    isLoading: VueTypes.bool.def(false),
    disabled: VueTypes.bool.def(false),
    outline: VueTypes.bool.def(false),
  },

  methods: {
    btnClicked() {
      this.$emit('btn-clicked');
    },
  },

};
</script>

<style lang="scss" scoped>
@import "@/theme/index.scss";

.btn {
  background: $brand;
  border-radius: calculateRem(8px);
  color: $white;
  min-height: calculateRem(45px);
  text-align: center;
  text-decoration: none;
  font-weight: bold;
  font-size: $fs-base;
  transition: all .15s linear;
  padding: calculateRem(5px) calculateRem(15px);
  cursor: pointer;
  border: none;
  &:hover {
    background: lighten($brand, 20%);
    transition: background .3s ease-in;
  }

  &:active {
    background-color: $brand;
  }

  &:disabled {
    cursor: not-allowed;
    background: $disabled;
  }

  &.outline {
    background: transparent;
    color: $white;
    border: calculateRem(1px) solid $white;
    &:hover {
      background: lighten($brand, 20%);
      border-color: inherit;
      color: $white;
      transition: background .3s ease-in;
    }
  }

  .content {
    margin: 0;
  }
}
</style>
