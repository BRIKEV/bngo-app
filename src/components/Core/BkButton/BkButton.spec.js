import { mount, shallowMount } from '@vue/test-utils';
import BkButton from '@/components/Core/BkButton/BkButton.vue';

describe('BkButton component', () => {
  it('Should be render correctly', () => {
    const wrapper = mount(BkButton);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Should has outline class', () => {
    const wrapper = shallowMount(BkButton, {
      propsData: { outline: true },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
