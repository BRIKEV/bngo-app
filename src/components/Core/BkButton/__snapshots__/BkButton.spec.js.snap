// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BkButton component Should be render correctly 1`] = `
<button
  class="btn"
>
  <!---->
   
  <p
    class="content"
  />
</button>
`;

exports[`BkButton component Should has outline class 1`] = `
<button
  class="btn outline"
>
  <beat-loader-stub
    color="white"
    margin="2px"
    size="10"
    sizeunit="px"
  />
   
  <p
    class="content"
  />
</button>
`;
